src/video.js

```diff
@@ -64,3 +69,59 @@ export const checkVideoBidSetup = hook('sync', function(bid, adUnit, videoMediaT
 
   return true;
 }, 'checkVideoBidSetup');
+
+export function registerVastTrackers(moduleType, moduleName, trackerFn) {
+  if (typeof trackerFn == 'function') { vastTrackers.push({'moduleType': moduleType, 'moduleName': moduleName, 'trackerFn': trackerFn}); }
+}
+
+export function insertVastTrackers(trackers, vastXml) {
+  const doc = new DOMParser().parseFromString(vastXml, 'text/xml');
+  const wrappers = doc.querySelectorAll('VAST Ad Wrapper, VAST Ad InLine');
+  try {
+    if (wrappers.length) {
+      wrappers.forEach(wrapper => {
+        if (trackers.get('impression')) {
+          trackers.get('impression').forEach(trackingUrl => {
+            const impression = doc.createElement('Impression');
+            impression.appendChild(doc.createCDATASection(trackingUrl));
+            wrapper.appendChild(impression)
+          });
+        }
+      });
+      vastXml = new XMLSerializer().serializeToString(doc);
+    }
+  } catch (error) {
+    logError('an error happened trying to insert trackers in vastXml');
+  }
+  return vastXml;
+}
+
+export function getVastTrackers(bid) {
+  let trackers = [];
+  vastTrackers.filter(
+    ({moduleType, moduleName, trackerFn}) => isActivityAllowed(ACTIVITY_REPORT_ANALYTICS, activityParams(moduleType, moduleName))
+  ).forEach(({trackerFn}) => {
+    let trackersToAdd = trackerFn(bid);
+    trackersToAdd.forEach(trackerToAdd => {
+      if (validVastTracker(trackers, trackerToAdd)) { trackers.push(trackerToAdd); }
+    });
+  });
+  const trackersMap = trackersToMap(trackers);
+  return (trackersMap.size ? trackersMap : null);
+};
+
+function validVastTracker(trackers, trackerToAdd) {
+  if (!trackerToAdd.hasOwnProperty('event') || !trackerToAdd.hasOwnProperty('url')) { return false; }
+  trackers.forEach(tracker => {
+    if (tracker['event'] == trackerToAdd['event'] && tracker['url'] == trackerToAdd['url']) { return false; }
+  });
+  return true;
+}
+
+function trackersToMap(trackers) {
+  let trackersMap = new Map();
+  trackers.forEach(tracker => {
+    if (!trackersMap.get(tracker['event'])) { trackersMap.set(tracker['event'], [tracker['url']]) } else { trackersMap.get(tracker['event']).push(tracker['url']); }
```

Comment:
@dgirardi any thoughts on code style ? I personally find `if` statements on the same line to be hard to read, and i find it even harder to read when the `else` statement is also on the same line.

Reply:
Agreed some line breaks would improve readability. 

I don't mind it too much though because it's not really poor style but poor formatting, and any half-decent editor can fix it with one click.

Reply:
Strange, I didn't wrote it this way. It's fixed.